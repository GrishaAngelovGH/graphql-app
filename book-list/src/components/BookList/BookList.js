import { useState } from "react"
import { gql, useQuery } from "@apollo/client"

import Badge from "react-bootstrap/Badge"
import Button from "react-bootstrap/Button"

import BookModal from "./BookModal"

const GET_BOOKS = gql`
  query GetBooks {
    books {
      isbn
      title
    }
  }
`

const BookList = () => {
  const [showBookModal, setShowBookModal] = useState(false)

  const { loading, data } = useQuery(GET_BOOKS)

  const handleShowBookModal = () => {
    setShowBookModal(!showBookModal)
  }

  return (
    <div className="row g-0">
      <div className="col-md-7 overflow-auto" style={{ height: 450 }}>
        <h3 className="text-center text-secondary fw-bold">Books</h3>

        {loading && <p className="text-center">Loading...</p>}

        {
          data?.books.map(v => (
            <Badge bg="primary" className="mx-2 mt-2 fs-6" key={v.isbn}>
              {v.title}
            </Badge>
          ))
        }

        {showBookModal && <BookModal showModal={showBookModal} onClose={handleShowBookModal} />}

        <Button
          variant="outline-success"
          className="fixed-bottom w-25"
          style={{ left: 20, bottom: 20 }}
          onClick={handleShowBookModal}
        >
          Add New Book
        </Button>
      </div>
    </div>
  )
}

export default BookList