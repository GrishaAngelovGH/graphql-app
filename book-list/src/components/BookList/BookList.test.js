import { gql } from '@apollo/client'
import { MockedProvider } from '@apollo/client/testing'
import { act, render } from '@testing-library/react'

import BookList from './BookList'

const mocks = [
  {
    request: {
      query: gql`
      query GetBooks {
        books {
          isbn
          title
        }
      }
    `
    },
    result: {
      data: {
        books: [
          { isbn: '1', title: 'Title 1' },
          { isbn: '2', title: 'Title 2' },
          { isbn: '3', title: 'Title 3' }
        ]
      }
    }
  }
]

test('should render BookList component', async () => {
  const view = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BookList />
    </MockedProvider>
  )

  await act(() => new Promise(resolve => setTimeout(resolve, 0)))

  expect(view).toMatchSnapshot()
})