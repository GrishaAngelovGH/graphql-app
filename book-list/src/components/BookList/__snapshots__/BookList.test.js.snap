// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BookList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0"
      >
        <div
          class="col-md-7 overflow-auto"
          style="height: 450px;"
        >
          <h3
            class="text-center text-secondary fw-bold"
          >
            Books
          </h3>
          <span
            class="mx-2 mt-2 fs-6 badge bg-primary"
          >
            Title 1
          </span>
          <span
            class="mx-2 mt-2 fs-6 badge bg-primary"
          >
            Title 2
          </span>
          <span
            class="mx-2 mt-2 fs-6 badge bg-primary"
          >
            Title 3
          </span>
          <button
            class="fixed-bottom w-25 btn btn-outline-success"
            style="left: 20px; bottom: 20px;"
            type="button"
          >
            Add New Book
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0"
    >
      <div
        class="col-md-7 overflow-auto"
        style="height: 450px;"
      >
        <h3
          class="text-center text-secondary fw-bold"
        >
          Books
        </h3>
        <span
          class="mx-2 mt-2 fs-6 badge bg-primary"
        >
          Title 1
        </span>
        <span
          class="mx-2 mt-2 fs-6 badge bg-primary"
        >
          Title 2
        </span>
        <span
          class="mx-2 mt-2 fs-6 badge bg-primary"
        >
          Title 3
        </span>
        <button
          class="fixed-bottom w-25 btn btn-outline-success"
          style="left: 20px; bottom: 20px;"
          type="button"
        >
          Add New Book
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
